dor/providers/utilities.py

```diff
@@ -0,0 +1,17 @@
+import re
+
+
+#
+# The perl regular expression
+#
+# s{[^\w+_\-\.]}{}gsm
+#
+# removes all characters from the input that are not
+# letters, digits, underscores, plus signs, hyphens, or dots,
+# treating the string in a global manner so that all instances are addressed.
+# The s and m modifiers have no effect in this particular expression
+# but are theoretically in place if you were to adapt this pattern
+# with line-sensitive logic.
+#
+def sanitize_basename(basename: str) -> str:
+    return re.sub(r'[^\w\+\-\.]', r'', basename.lower())
```

Comment:
There's an alternative syntax we could use where you compile the regex first. For example:

```py
pattern = re.compile(r'[^\w\+\-\.]')

def sanitize_basename(basename: str) -> str:
    return pattern.sub(basename.lower())
```

Not sure it changes much, but I've used it before, thought I'd mention it.